src/Symfony/Component/Form/Extension/Csrf/EventListener/CsrfValidationListener.php

```diff
@@ -59,14 +60,15 @@ public function preSubmit(FormEvent $event)
         if ($form->isRoot() && $form->getConfig()->getOption('compound') && !$postRequestSizeExceeded) {
             $data = $event->getData();
 
-            if (!isset($data[$this->fieldName]) || !$this->tokenManager->isTokenValid(new CsrfToken($this->tokenId, $data[$this->fieldName]))) {
+            $csrfToken = new CsrfToken($this->tokenId, $data[$this->fieldName]);
```

Comment:
`$data[$this->fieldName] ?? null` and perhaps replace `!isset()` with `'' === $csrtfToken->getValu()` below to be sure

Reply:
It doesn't feel right to me this way. Waiting for @xabbuh feedback.

Reply:
@ro0NL has a good point here. Just having the `CsrfToken` class as the cause sounds good to me.